clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -397,10 +397,10 @@ protected void onJoinComplete(int generation,
                     "\tCurrent owned partitions:                  {}\n" +
                     "\tAdded partitions (assigned - owned):       {}\n" +
                     "\tRevoked partitions (owned - assigned):     {}\n",
-                assignedPartitions,
-                ownedPartitions,
-                addedPartitions,
-                revokedPartitions
+                Utils.join(assignedPartitions.stream().sorted().toArray(), ", "),
+                Utils.join(ownedPartitions.stream().sorted().toArray(), ", "),
+                Utils.join(addedPartitions.stream().sorted().toArray(), ", "),
+                Utils.join(revokedPartitions.stream().sorted().toArray(), ", ")
```

Comment:
Actually, the `assignedPartitions`, `addedPartitions`, and `revokedPartitions`, and the following `invokePartitionsRevoked`, `invokePartitionsAssigned` are coming from the `assignment.partitions()`(or the `assignedPartitions`). We can sort it first, and the rest of the partitions will be sorted.

Just the `ownedPartitions` might still need to sort itself.